__tests__/profile/charts/barchart.test.js

```diff
@@ -367,9 +367,79 @@ describe('Test downloadable barchart', () => {
         primary_group: 'age',
         groups: [ { name: 'age' } ]
       }
-      let vegaDownloadSpec = configureBarchartDownload([], metadata, config, annotations);
+      let vegaDownloadSpec = configureBarchartDownload([], [], metadata, config, annotations);
       let view = renderVegaHeadless(vegaDownloadSpec);
       await view.runAsync()
       expect(view.signal('source')).toBe('Source : test');
     });
 });
+
+
+describe('configureBarchart for missing data', () => {
+    document.body.innerHTML = html;
+
+    let data, metadata, config;
+    beforeEach(() => {
+        metadata = {
+            primary_group: "education_level",
+            groups: [{name: "education_level"}, {name: "gender"}]
+        },
+            data = [
+                {age: 15, gender: 'male', count: 1, education_level: "Grade9"},
+                {age: 12, gender: 'female', count: 3, education_level: "Grade10"},
+                {age: 14, gender: 'male', count: 2, education_level: "Grade9"}
+            ]
```

Comment:
```suggestion
        }
        data = [
            {age: 15, gender: 'male', count: 1, education_level: "Grade9"},
            {age: 12, gender: 'female', count: 3, education_level: "Grade10"},
            {age: 14, gender: 'male', count: 2, education_level: "Grade9"}
        ]
```
I think this can be tidied up